crates/strategies/blend-strategies/src/auction_manager.rs

```diff
@@ -351,7 +357,14 @@ fn get_fill_info(min_profit: i128, lot_value: i128, bid_value: i128) -> (i128, i
             mod_bid_value -= step_bid_value;
         }
         let profit = mod_lot_value - mod_bid_value;
-        if profit >= min_profit {
+        // check if our profit has reached the minimum profit percentage
+        if profit > 0
+            && profit
+                .fixed_div_floor(mod_bid_value, SCALAR_7)
+                .unwrap_or(1000000)
+                >= min_profit_pct
+        {
```

Comment:
Assuming the `unwrap_or` was supposed to be 100%?

This being said, the case where `mod_bid_value == 0` (i.e. block 400+, or potentially block 399 with a very small auction), we might be better off checking that case specifically. In the event of any other failure during `fixed_div_floor`, we might be better off not returning from the for loop (I.E. `unwrap_or(0)`)

Reply:
yeah that's better